sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationChecker.scala

```diff
@@ -509,7 +509,24 @@ object UnsupportedOperationChecker extends Logging {
           throwError("Sampling is not supported on streaming DataFrames/Datasets")
 
         case Window(_, _, _, child) if child.isStreaming =>
-          throwError("Non-time-based windows are not supported on streaming DataFrames/Datasets")
+          val windowExpression = subPlan.asInstanceOf[Window].windowExpressions
+          val windowFuncs = windowExpression.flatMap { e =>
+            e.collect {
+              case we: WindowExpression =>
+                s"'${we.windowFunction}' as column '${e.toAttribute.sql}'"
+            }
+          }.mkString(", ")
+          val windowSpec = windowExpression.flatMap { e =>
+            e.collect {
+              case we: WindowExpression => we.windowSpec.sql
+            }
+          }.mkString(", ")
+          throw new AnalysisException(
+            s"Window function is not supported in $windowFuncs on streaming DataFrames/Datasets. " +
+            "Structured Streaming only supports time-window aggregation using the `window` " +
+            s"function. (window specification: '$windowSpec')",
```

Comment:
Could you introduce an error class with this error format, and put it to `error-classes.json`, please. Also cc @HeartSaVioR for checking semantic.

Reply:
@siying 
The change itself seems to be OK.

We are going to apply error class framework on new attempt of touching exception. Although we haven't made any progress on UnsupportedOperationChecker, we could consider this as a starter.

Please look into the recent PR on error class migration in SS, especially `error-classes.json` and `QueryExecutionErrors.scala`.
https://github.com/apache/spark/pull/41205

Here is a guide doc for error class framework.
https://github.com/apache/spark/blob/master/core/src/main/resources/error/README.md

Please go through and apply it. Thanks!